### ISSUE TITLE
error on installation of module accountancy_install

### ISSUE THREAD

**Issue by ThijsvOers** [onesteinbv/addons-container#43]
on installation of the module accountancy_install I get the following error. 

RPC_ERROR
Odoo Server Error
Traceback (most recent call last):
  File "/odoo/src/odoo/odoo/modules/registry.py", line 90, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/odoo/src/odoo/odoo/modules/loading.py", line 462, in load_modules
    env.flush_all()
  File "/odoo/src/odoo/odoo/api.py", line 734, in flush_all
    self[model_name].flush_model()
  File "/odoo/src/odoo/odoo/models.py", line 5559, in flush_model
    self._flush(fnames)
  File "/odoo/src/odoo/odoo/models.py", line 5632, in _flush
    process(model, id_vals)
  File "/odoo/src/odoo/odoo/models.py", line 5582, in process
    model.browse(ids)._write(vals)
  File "/odoo/src/odoo/odoo/models.py", line 3834, in _write
    cr.execute(query, params + [sub_ids])
  File "/odoo/src/odoo/odoo/sql_db.py", line 313, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.SerializationFailure: could not serialize access due to concurrent update

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/odoo/src/odoo/odoo/http.py", line 1583, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/odoo/src/odoo/odoo/service/model.py", line 134, in retrying
    result = func()
  File "/odoo/src/odoo/odoo/http.py", line 1610, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/odoo/src/odoo/odoo/http.py", line 1807, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/odoo/src/odoo/odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "/odoo/src/odoo/odoo/http.py", line 696, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/odoo/src/odoo/addons/web/controllers/dataset.py", line 46, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/odoo/src/odoo/addons/web/controllers/dataset.py", line 33, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/odoo/src/odoo/odoo/api.py", line 461, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/odoo/src/odoo/odoo/api.py", line 448, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-74>", line 2, in button_immediate_install
  File "/odoo/src/odoo/odoo/addons/base/models/ir_module.py", line 76, in check_and_log
    return method(self, *args, **kwargs)
  File "/odoo/src/odoo/odoo/addons/base/models/ir_module.py", line 491, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/odoo/src/odoo/odoo/addons/base/models/ir_module.py", line 615, in _button_immediate_function
    registry = modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "<decorator-gen-14>", line 2, in new
  File "/odoo/src/odoo/odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "/odoo/src/odoo/odoo/modules/registry.py", line 92, in new
    odoo.modules.reset_modules_state(db_name)
  File "/odoo/src/odoo/odoo/modules/loading.py", line 637, in reset_modules_state
    cr.execute(
  File "/odoo/src/odoo/odoo/sql_db.py", line 313, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.SerializationFailure: could not serialize access due to concurrent update

The above server error caused the following client error:
RPC_ERROR: Odoo Server Error
    at makeErrorFromResponse (https://dev.flavours.k8s.onestein.eu/web/assets/21-04fb39e/web.assets_backend.min.js:970:163)
    at XMLHttpRequest.<anonymous> (https://dev.flavours.k8s.onestein.eu/web/assets/21-04fb39e/web.assets_backend.min.js:978:13)

**Comment by ThijsvOers** [comment]
after some change by Andrea, I get this error on activating the account_install module: RPC_ERROR
Odoo Server Error
Traceback (most recent call last):
  File "/odoo/src/odoo/odoo/modules/registry.py", line 90, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/odoo/src/odoo/odoo/modules/loading.py", line 462, in load_modules
    env.flush_all()
  File "/odoo/src/odoo/odoo/api.py", line 734, in flush_all
    self[model_name].flush_model()
  File "/odoo/src/odoo/odoo/models.py", line 5559, in flush_model
    self._flush(fnames)
  File "/odoo/src/odoo/odoo/models.py", line 5632, in _flush
    process(model, id_vals)
  File "/odoo/src/odoo/odoo/models.py", line 5582, in process
    model.browse(ids)._write(vals)
  File "/odoo/src/odoo/odoo/models.py", line 3834, in _write
    cr.execute(query, params + [sub_ids])
  File "/odoo/src/odoo/odoo/sql_db.py", line 313, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.SerializationFailure: could not serialize access due to concurrent update


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/odoo/src/odoo/odoo/http.py", line 1583, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/odoo/src/odoo/odoo/service/model.py", line 134, in retrying
    result = func()
  File "/odoo/src/odoo/odoo/http.py", line 1610, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/odoo/src/odoo/odoo/http.py", line 1807, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/odoo/src/odoo/odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "/odoo/src/odoo/odoo/http.py", line 696, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/odoo/src/odoo/addons/web/controllers/dataset.py", line 46, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/odoo/src/odoo/addons/web/controllers/dataset.py", line 33, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/odoo/src/odoo/odoo/api.py", line 461, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/odoo/src/odoo/odoo/api.py", line 448, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-74>", line 2, in button_immediate_install
  File "/odoo/src/odoo/odoo/addons/base/models/ir_module.py", line 76, in check_and_log
    return method(self, *args, **kwargs)
  File "/odoo/src/odoo/odoo/addons/base/models/ir_module.py", line 491, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/odoo/src/odoo/odoo/addons/base/models/ir_module.py", line 615, in _button_immediate_function
    registry = modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "<decorator-gen-14>", line 2, in new
  File "/odoo/src/odoo/odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "/odoo/src/odoo/odoo/modules/registry.py", line 92, in new
    odoo.modules.reset_modules_state(db_name)
  File "/odoo/src/odoo/odoo/modules/loading.py", line 637, in reset_modules_state
    cr.execute(
  File "/odoo/src/odoo/odoo/sql_db.py", line 313, in execute
    res = self._obj.execute(query, params)
psycopg2.errors.DeadlockDetected: deadlock detected
DETAIL:  Process 24463 waits for ShareLock on transaction 22555; blocked by process 24468.
Process 24468 waits for ShareLock on transaction 22556; blocked by process 24463.
HINT:  See server log for query details.
CONTEXT:  while updating tuple (49,17) in relation "ir_module_module"


The above server error caused the following client error:
RPC_ERROR: Odoo Server Error
    at makeErrorFromResponse (https://dev.flavours.k8s.onestein.eu/web/assets/21-04fb39e/web.assets_backend.min.js:970:163)
    at XMLHttpRequest.<anonymous> (https://dev.flavours.k8s.onestein.eu/web/assets/21-04fb39e/web.assets_backend.min.js:978:13)

**Comment by astirpe** [comment]
I didn't change anything

**Comment by astirpe** [comment]
The "concurrent update" issue and the "deadlock detected" issue are both caused by the locking on database for a relatively long time. The locking on db usually happens when making an update of a large number of module or when installing modules with a lot of dependencies, like in this case the accountancy_install module.

To mitigate the problem, I proposed to Dennis to pause the cron jobs during updates and also showing a temporary "Update in progress" page while updating, so the user would not interfere.

Closing this issue, since there's nothing we can fix on the modules of this repository.
